core/textile/sync/queue.go

```diff
@@ -0,0 +1,132 @@
+package sync
+
+import (
+	"container/list"
+	"encoding/json"
+	"fmt"
+
+	"github.com/FleekHQ/space-daemon/log"
+)
+
+const QueueStoreKey = "TextileSyncTaskQueue"
+
+type marshalledQueue struct {
+	queueAsSlice     []*Task          `json:"queueAsSlice"`
+	fileQueueAsSlice []*Task          `json:"fileQueueAsSlice"`
+	hashMap          map[string]*Task `json:"queueHashMap"`
+}
+
+func (s *synchronizer) enqueueTask(task *Task, queue *list.List) {
+	queue.PushBack(task)
```

Comment:
@dmerrill6 

Do you think we can simplify `container/list` to just a slice?

Reply:
I chose using a linked list because it's perfect to represent a queue, as queue/dequeue operations are O(1) and memory efficient. Not sure really how Go handles slices internally when doing stuff like `slice = slice[:-1]` but I rather have control.